server_src/services/database/server-settings.service.js

```diff
@@ -0,0 +1,294 @@
+const ServerSettingsDB = require("../../models/ServerSettings.js");
+
+const { isEmpty } = require("lodash");
+
+class ServerSettings {
+  currentServerSettings;
+  /**
+   * Initialises server settings on boot, reboot and makes sure all required values are defined. Creates cache of current values.
+   * @returns {String} String stating success or failure
+   */
+  async init() {
+    let currentSettings;
+    try {
+      currentSettings = await this.returnDatabaseValues();
+      if (currentSettings.length < 1) {
+        let defaultServerSettings = new ServerSettingsDB({});
+        await defaultServerSettings.save();
+        this.currentServerSettings = defaultServerSettings;
+        return "Default server settings have been created";
+      } else {
+        // Re-save settings to make sure defaults are available...
+        await currentSettings[0].save();
+        this.currentServerSettings = currentSettings[0];
+        return "Server settings already exist, loaded existing values...";
+      }
+    } catch (e) {
+      if (e.message.includes("command find requires authentication")) {
+        throw "Database authentication failed.";
+      } else {
+        throw "Database connection failed.";
+      }
+    }
+  }
+  /**
+   * Resets server settings to default and updates the internal cache
+   */
+  async resetServerSettingsToDefault() {
+    const currentDatabaseSettings = await this.returnDatabaseValues({});
+    await currentDatabaseSettings.deleteMany({});
+    await currentDatabaseSettings.save();
+    this.currentServerSettings = currentDatabaseSettings;
+  }
+  /**
+   * Checks the database for the server settings, can be supplied with filter and options to specify mongoose options/filter.
+   * @param {Object} filter object to filter the database results
+   * @param {Object} options object to change the database options
+   * @returns {Object} Object containing the filtered settings / full settings
+   */
+  returnDatabaseValues({ filter = {}, options = {} } = {}) {
+    return ServerSettingsDB.find(filter, {}, options);
+  }
+  /**
+   * Returns the entire settings object from catch
+   * @returns {Object} Object from cache for all of OctoFarms settings
+   */
+  get entireSettingsObject() {
+    return this.currentServerSettings;
+  }
+
+  /**
+   * Returns the polling settings for OctoPrints websocket connection
+   * @returns {Object} Object from cache for OctoPrints websocket connection
+   */
+  get octoPrintWebsocketPollingSettings() {
+    return this.currentServerSettings["onlinePolling"];
```

Comment:
Can you explain to me the reason for these getter functions? I can think of many, just interested in hearing your thoughts on it, to be able to check that this stays manageable when it scales up